### ISSUE TITLE
Importer should explicitly require Account ID

### ISSUE THREAD

**Issue by RickyRomero** [matomo-org/plugin-GoogleAnalyticsImporter#10]
Plugin: 1.0.2
Matomo: 3.12.0-rc2

Hi,

I tried to set this plugin up on behalf of my wife, who is trying to migrate off of G Suite. I followed the guide [on this page](https://matomo.org/docs/google-analytics-importer/), but after setting up the OAuth permissions and credentials, when I clicked `Schedule an Import`, the button did nothing. I opened up the web inspector for clues and only found a single generic-looking Angular error in the JS console when the page loaded.

<img width="1110" alt="Screen_Shot_2019-10-27_at_7 02 10_PM" src="https://user-images.githubusercontent.com/233891/67646650-63b09d00-f8ec-11e9-8563-e32c16b2554d.png">

I tried resetting the credentials and reuploading them several times with no effect. I also changed the secrets and broadened permissions on the Google Cloud API end, which didn't help.

![Screen_Shot_2019-10-27_at_6 47 00_PM](https://user-images.githubusercontent.com/233891/67646732-b8541800-f8ec-11e9-8008-dc5ef426b9c3.png)
<img width="1355" alt="Screen_Shot_2019-10-27_at_6 46 41_PM" src="https://user-images.githubusercontent.com/233891/67646789-ecc7d400-f8ec-11e9-9a4b-5b317ebdab47.png">

Next, I tried following the instructions for running the import from the console, which got me far enough to see that it's a permissions problem. But I don't understand why, considering that I followed the tutorial to the letter...

<img width="1271" alt="Screen_Shot_2019-10-27_at_6 45 51_PM" src="https://user-images.githubusercontent.com/233891/67646864-36b0ba00-f8ed-11e9-8f0c-55ed24bfe055.png">

(Narrowing the date range of the import to one month produces the same result.)

I'm stumped. Do G Suite accounts not have access to this info? Is this failing because I haven't enabled API access of some sort in the Google Analytics UI itself?

**Comment by diosmosis** [comment]
For the first error, can you ensure the plugin is activated, then run `./console cache:clear`, then clear your browser cache and reload the page? The error is stating the controller doesn't exist, which suggests the JS being loaded doesn't include the new JS.

For the second error, can you run the command w/ `-vvv`? This should display a stack trace. I'm not sure exactly why the permissions error is being thrown, but maybe we can see what API call is resulting in it.

**Comment by RickyRomero** [comment]
@diosmosis Thanks for the fast reply and diagnostics!

I tried your second suggestion first, in case clearing the cache fixed the problem. [Here is the stack trace](https://gist.github.com/RickyRomero/79c230bd00f6aa16879e9990a3624bf8).

As for clearing the cache... It didn't end up fixing the permissions bug, but it did fix the UI error. Here are screenshots of the settings I used and what happened after.

![image](https://user-images.githubusercontent.com/233891/67726056-da5ba200-f9a1-11e9-955b-ed89c7d77610.png)
![image](https://user-images.githubusercontent.com/233891/67726078-ec3d4500-f9a1-11e9-9f80-9ddc2b22a06f.png)

I'm wondering if perhaps the view ID I copied is the wrong number. The UI says it should be 7 digits long, but the one I found is 8.

**Comment by diosmosis** [comment]
The view ID should be under 'Views' in GA in this popup:

![image](https://user-images.githubusercontent.com/125140/67795573-b0ef5480-fa3b-11e9-963e-1a6e4db80836.png)

Did you try running the CLI command w/ the `-vvv` option? That will provide a stack trace to show which specific API request to GA is failing.

**Comment by RickyRomero** [comment]
@diosmosis Thanks for that. I just checked that menu and verified that I used the correct IDs.

[Here's a link to the gist for the stack trace](https://gist.github.com/RickyRomero/79c230bd00f6aa16879e9990a3624bf8).

**Comment by diosmosis** [comment]
@RickyRomero Sorry, I see now I missed the original link for the stack trace.

Ok, it's definitely something on the Google side of things, though I'm not sure what exactly. Can you ensure that the account ID in GA is `16474406`? Is the user you're using to access the analytics account the user that owns the analytics property? Also if you made a change to the oauth consent screen after uploading the client secret to Matomo, you'd have to re-import the credentials (Google would change the secret I believe). The only other thing I can think of is the "Internal" radio button in the consent screen. Would it be possible to check if it works when "Public" is selected?

**Comment by RickyRomero** [comment]
These appear to be the relevant IDs for Heather's G Suite account:

**Account ID:** `104121435`
**Property ID:** `UA-16474406-1`
**View ID:** `32780573`

I started writing a long response giving lots of thorough detail about the issue and how I wasn't getting anywhere. Long story short, I stumbled across this function as I tried to get more clues as to what was happening.

https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/blob/f82a52c6fa46c68f8d90a5484008d0c80887a60e/Commands/ImportReports.php#L238-L245

It appears that the plugin assumes that the account ID occurs within the property ID. This is not the case for Heather's G Suite account, as you can see above.

When I entered the account ID manually, using the command line, it began hitting the Google Analytics API without errors.

```
[akhenaten]$ ./console -vvv googleanalyticsimporter:import-reports --account=104121435 --property=UA-16474406-1 --view=32780573 --dates=2010-01-01,2019-10-28
DEBUG [2019-10-31 04:29:38] 23168  Website 2 tracker cache was re-created.
Created new site with ID = 2.
WARNING [2019-10-31 04:29:38] 23168  The CustomDimensions plugin is not activated or present, skipping goal import.
INFO [2019-10-31 04:29:38] 23168  Using existing custom variable slots.
Importing reports for date range 2010-01-01 - 2019-10-28 from GA view 32780573.
INFO [2019-10-31 04:29:38] 23168  Importing data for GA View 32780573 for date 2010-01-01...
DEBUG [2019-10-31 04:29:39] 23168  Importing data for the VisitsSummary plugin.
INFO [2019-10-31 04:29:39] 23168  Found 0 sessions for 2010-01-01, skipping rest of plugins for this day.
INFO [2019-10-31 04:29:39] 23168  Importing data for GA View 32780573 for date 2010-01-02...
DEBUG [2019-10-31 04:29:39] 23168  Importing data for the VisitsSummary plugin.
INFO [2019-10-31 04:29:39] 23168  Found 0 sessions for 2010-01-02, skipping rest of plugins for this day.
INFO [2019-10-31 04:29:39] 23168  Importing data for GA View 32780573 for date 2010-01-03...
DEBUG [2019-10-31 04:29:39] 23168  Importing data for the VisitsSummary plugin.
INFO [2019-10-31 04:29:40] 23168  Found 0 sessions for 2010-01-03, skipping rest of plugins for this day.
INFO [2019-10-31 04:29:40] 23168  Importing data for GA View 32780573 for date 2010-01-04...
DEBUG [2019-10-31 04:29:40] 23168  Importing data for the VisitsSummary plugin.
INFO [2019-10-31 04:29:40] 23168  Found 0 sessions for 2010-01-04, skipping rest of plugins for this day.
[snip]
```

**Comment by diosmosis** [comment]
Thanks for debugging the issue and finding the root cause! I'll be able to fix this issue after the weekend.

Note: the account ID is only assumed in the command when `--account` is not specified. When the import job is restarted the next day (which it should even if you start the import from the command line), the account you entered in the option will be used.

**Comment by diosmosis** [comment]
@RickyRomero this should be fixed in version 1.0.3, let me know if you experience any more issues.

**Comment by RickyRomero** [comment]
Thank you so much! 🎉
